### PR TITLE
Fix settings page

### PR DESCRIPTION
**Pull request by charleshenryhugo**
PR #139 introduced some issues :
1. When a customer enters in the settings page and then clicks `保存` button, the admin-order-list page is returned.
2. The switch button moves slowly.

Some changes were made in this PR to fix the issues above.
1. `保存`button removed. 
2. The database update is done when admin clicks the switch button,

video link:
https://youtu.be/BDZHRYCcc1Q

### CODE REVIEW

File: lib/screens/settings_page.dart [hunk 1]
```diff
@@ -61,9 +61,21 @@ class _SettingsPageState extends State<SettingsPage> {
...
-          Text('Customer Settings Listing'),
+          // TODO customer settings listing
+          Container(),
         ],
       ),
     );
   }
+
+  Future<void> _saveCurrentSignedInUserData() async {
+    if (currentSignedInUser.isAdmin()) {
```
Comment (dragonfly91) [inline lib/screens/settings_page.dart]: Can we do this update in the `AdminSettingsListing` widget?

File: lib/widgets/admin_settings_listing.dart [hunk 2]
```diff
@@ -1,15 +1,19 @@
 import 'package:flutter/material.dart';
-import 'package:badiup/constants.dart' as constants;
 import 'package:badiup/colors.dart';
 import 'package:badiup/sign_in.dart';
 
+bool taxInclusiveBoolValue;
```
Comment (dragonfly91) [inline lib/widgets/admin_settings_listing.dart]: Based on the previous comment, this could be a private variable

**Review by dragonfly91 — CHANGES_REQUESTED** [review]
Left some refactoring comments.

**Comment by dragonfly91** [comment]
Also, can you post a video about how the switch button responds with your changes? @charleshenryhugo

File: lib/screens/settings_page.dart (hunk 1 above)
Comment (charleshenryhugo) [reply lib/screens/settings_page.dart]: I am stuck at how to call the functions of `AdminSettingsListing` widget from `SettingsPage` widget.

Comment (dragonfly91) [reply lib/screens/settings_page.dart]: I don’t think you have to call any functions. How about this implementation?
In `settings_page.dart`:

```
Future<void> _saveCurrentSignedInUserData() async {
    await db
        .collection(constants.DBCollections.users)
        .document(currentSignedInUser.email)
        .updateData(currentSignedInUser.toMap());
}
```

In `admin_settings_listing.dart`:

```
...
          Switch(
            activeColor: Colors.white,
            activeTrackColor: paletteForegroundColor,
            value: currentSignedInUser.setting.taxInclusive,
            onChanged: (bool value) {
              setState(() {
                currentSignedInUser.setting.taxInclusive = value;
              });
            },
          )
...
```

**Comment by dragonfly91** [comment]
https://github.com/GospelAid/badiup/issues/150 need not be fixed if this PR goes thru.

File: lib/screens/settings_page.dart (hunk 1 above)
Comment (charleshenryhugo) [reply lib/screens/settings_page.dart]: Wanted to write as you suggested!
However, if the user clicks the switch button and then just goes back to the previous page without clicking the `保存` button, 
only the `currentSignedInUser.setting.taxInclusive` will be updated, while the database keeps the old record.

`currentSignedInUser.setting.taxInclusive` and the database should be updated within the same function, i think.

**Review by dragonfly91 — APPROVED** [review]

**Pull request merged**